test/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/CheckClaimantDetailsControllerSpec.scala

```diff
@@ -627,6 +635,102 @@ class CheckClaimantDetailsControllerSpec
       isMandatoryDataAvailable(fillingOutClaim) shouldBe false
 
     }
+  }
+
+  "The address lookup" should {
+
+    "start successfully" in forAll(journeys) { journey =>
+      val lookupUrl = sample[URL]
+      val session   = getSessionWithPreviousAnswer(None, None, Some(toSelectNumberOfClaims(journey)))._1
+
+      inSequence {
+        mockAuthWithNoRetrievals()
+        mockGetSession(session.copy(journeyStatus = None))
+        mockAddressLookupInitiation(Right(lookupUrl))
+      }
+
+      checkIsRedirect(
+        changeAddress(journey),
+        lookupUrl.toString
+      )
+    }
+
+    "fail to start once error response received downstream ALF service" in forAll(journeys) { journey =>
+      val session = getSessionWithPreviousAnswer(None, None, Some(toSelectNumberOfClaims(journey)))._1
+
+      inSequence {
+        mockAuthWithNoRetrievals()
+        mockGetSession(session.copy(journeyStatus = None))
+        mockAddressLookupInitiation(Left(Error("Request was not accepted")))
+      }
+
+      checkIsTechnicalErrorPage(changeAddress(journey))
+    }
+
+    "update an address once complete" in forAll(journeys) { journey =>
+      val id           = sample[UUID]
+      val address      = sample[ContactAddress]
+      val acc14        = generateAcc14WithAddresses()
+      val (session, _) = getSessionWithPreviousAnswer(
+        Some(acc14),
+        Some(DeclarantTypeAnswer.Importer),
+        Some(toSelectNumberOfClaims(journey))
+      )
+
+      inSequence {
+        mockAuthWithNoRetrievals()
+        mockGetSession(session)
+        mockAddressRetrieve(Right(address))
+        mockStoreSession(Right(()))
+      }
+
+      checkIsRedirect(
+        updateAddress(journey, Some(id)),
+        routes.CheckClaimantDetailsController.show(journey)
+      )
```

Comment:
are you actually checking the session store is updated? (and the same for following tests)

Reply:
Yes, I mocking it by `mockStoreSession(Right(()))`, at least checking whether it been called